playbooks/openstack-cloud-controller-manager-acceptance-test-lb-octavia-nested/playbook-in-vm/run.yaml

```diff
@@ -0,0 +1,193 @@
+- include: ../../kubernetes-common/playbooks/add_vm_to_inventory.yaml
+
+- name: Prepare for local cluster
+  hosts: "{{ server_name }}"
+  become: yes
+  gather_facts: no
+  tasks:
+    - shell:
+        cmd: |
+          set -e
+          set -x
+          cat << EOF >> ./openrc
+          export OS_PROJECT_DOMAIN_ID=default
+          export OS_REGION_NAME=RegionOne
+          export OS_USER_DOMAIN_ID=default
+          export OS_PROJECT_NAME=admin
+          export OS_IDENTITY_API_VERSION=3
+          export OS_PASSWORD=secretadmin
+          export OS_AUTH_TYPE=password
+          export OS_AUTH_URL="http://{{ devstack_ip }}/identity"
+          export OS_USERNAME=admin
+          export OS_TENANT_NAME=admin
+          export OS_VOLUME_API_VERSION=2
+          EOF
+          source ./openrc
+
+          if [[ ! -d "/etc/kubernetes/" ]]; then
+              sudo mkdir -p /etc/kubernetes/
+          fi
+          chown ubuntu /etc/kubernetes/
+
+          cat << EOF >> /etc/kubernetes/cloud-config
+          [Global]
+          domain-id = default
+          tenant-id = "{{ project_id }}"
+          auth-url = ${OS_AUTH_URL}
+          password = secretadmin
+          username = admin
+          region = RegionOne
+
+          [LoadBalancer]
+          floating-network-id = "{{ floating_network_id }}"
+          subnet-id = "{{ subnet_id }}"
+
+          [BlockStorage]
+          bs-version = v2
+          EOF
+          # Install Make
+          apt-get install make -y
+          # Install docker
+          if ! dpkg -s "docker-engine" > /dev/null 2> /dev/null; then
+              sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D || true
+              sudo apt-key adv --keyserver hkp://pgp.mit.edu:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D || true
+              sudo apt-add-repository 'deb http://apt.dockerproject.org/repo ubuntu-xenial main'
+              sudo apt-get update -y
+              sudo apt-cache policy docker-engine
+              sudo apt-get install -y docker-engine=1.12.6-0~ubuntu-xenial
+              sudo cat /lib/systemd/system/docker.service
+              sudo sed -r -i "s|(ExecStart)=(.+)|\1=\2 --iptables=false|" /lib/systemd/system/docker.service
+              sudo cat /lib/systemd/system/docker.service
+              sudo systemctl daemon-reload
+              sudo systemctl restart docker
+              sudo systemctl status docker
+              sudo ifconfig -a
+          fi
+          docker --version
+          # Get the latest stable version of kubernetes
+          export K8S_VERSION=$(curl -sS https://storage.googleapis.com/kubernetes-release/release/stable.txt)
```

Comment:
We compile k8s from source code, these two line seems not necessary.

Reply:
done